smart-contract-verifier/smart-contract-verifier-proto/proto/smart-contract-verifier.proto

```diff
@@ -0,0 +1,131 @@
+  map<string, string> libraries = 7;
+}
+
+message VerifySolidityStandardJsonRequest {
+  /// If present, used to compare local compilation result with
+  optional string creation_tx_input = 1;
+  /// If creation_tx_input is not present, used to compare local compilation result with
+  string deployed_bytecode = 2;
+  /// Compiler version used to compile the contract
+  string compiler_version = 3;
+  /// https://docs.soliditylang.org/en/latest/using-the-compiler.html#input-description
+  string input = 4;
+}
+
+
+message VerifyVyperMultiPartRequest {
+  /// If present, used to compare local compilation result with
+  optional string creation_tx_input = 1;
+  /// If creation_tx_input is not present, used to compare local compilation result with
+  string deployed_bytecode = 2;
+  /// Compiler version used to compile the contract
+  string compiler_version = 3;
+  /// Source file name to the actual source code
+  map<string, string> sources = 6;
+  /// Version of the EVM to compile for
+  string evm_version = 4;
+}
+
+message VerifyViaSourcifyRequest {
+  /// Address of the contract to be verified
+  string address = 1;
+  /// The chain (network) the contract was deployed to
+  /// (https://docs.sourcify.dev/docs/api/chains/)
+  string chain = 2;
+  /// Files required for verification (see Sourcify Api)
+  map<string, string> files = 3;
+  // (optional) see Sourcify Api
+  optional int32 chosenContract = 4;
+}
+
+message VerifyResponse {
+  string message = 1;
+  /// Status of 0 indicates successful verification
+  int32 status = 2;
+
+  message Result {
+    /// The name of the file verified contract was located at
+    string file_name = 1;
+    /// The name of the contract which was verified
+    string contract_name = 2;
+    /// Compiler version used for verification
+    string compiler_version = 3;
+    /// Source files given for verification
+    map<string, string> sources = 4;
+    /// Version of the EVM contract was compile for
+    string evm_version = 5;
+    /// (optional) WARNING: Before version 0.8.6 omitting the 'enabled' key was not equivalent to setting
+    /// it to false and would actually disable all the optimizations.
+    optional bool optimizations = 6;
+    /// (optional) Specify number of optimizer runs, if optimizations are enabled
+    optional int32 optimization_runs = 7;
+    /// Addresses of the libraries
+    map<string, string> contract_libraries = 8;
+    /// Raw settings pushed submitted to the compiler on local compilation
+    /// (https://docs.soliditylang.org/en/v0.8.17/using-the-compiler.html#input-description)
+    string compiler_settings = 9;
+    /// Constructor arguments used for deploying verified contract
+    /// (exists only for creation inputs)
+    optional string constructor_arguments = 10;
+    /// Contract abi (https://docs.soliditylang.org/en/latest/abi-spec.html?highlight=abi#json);
+    /// (does not exist for Yul contracts)
+    optional string abi = 11;
+
+    message bytecode_part {
```

Comment:
oneof?

Reply:
We want bytecode part to be encoded as 
```
{
  "local_creation_input_parts": [
    {
        "type": "main",
        "data": "0xdafda"
    }
  ]
}
```
to be backwards compatible. Oneof seems not to be an option in that case

Reply:
what's wrong with this?
```json
{ "local_creation_input_parts": [ 
  { "main": "0xdafda" }
 ] }
```

Reply:
We've just implemented it that way in https://github.com/blockscout/blockscout-rs/pull/205 for Http server, as it was easier to implement serde serialization. And I've tried to make new API to correspond to the previous one. So the question was not over consideration here.

Furthermore, in my opinion, explicit tag is better in the sense that it makes the semantic of the field clearer. In that case, it is explicitly mentioned that "main" is the value of key "type", while "0xdafda" is the actual data. While in your approach "0xdafda" seems to be a value for the key "main" and it is more difficult to understand what is happening here without the context, imo.